workers/common/src/main/kotlin/common/env/EnvironmentService.kt

```diff
@@ -164,11 +178,38 @@ class EnvironmentService(
     /**
      * Perform the required steps to set up authentication information for the current worker execution in the current
      * ORT run as defined by the given [context]. Load the infrastructure services associated with the current run from
-     * the database and process their credentials. This function can be used by workers running after the Analyzer,
-     * which has initialized the required information.
+     * the database. Given the secret names in [DynamicInfrastructureService.usernameSecretName] and
+     * [DynamicInfrastructureService.passwordSecretName], find these secrets and continue to process them.
+     * This function can be used by workers running after the Analyzer, which has initialized the required information.
      */
     suspend fun setupAuthenticationForCurrentRun(context: WorkerContext) {
-        setupAuthentication(context, infrastructureServiceRepository.listForRun(context.ortRun.id))
+        val dynamicInfrastructureServices = dynamicInfrastructureServiceRepository.listForRun(context.ortRun.id)
+
+        val infrastructureServices = dynamicInfrastructureServices.map { service ->
```

Comment:
Doing this resolving one by one may be inefficient, also if there are services that share secrets (e.g. for username). In the worker context, functionality is available to resolve secrets in parallel - maybe something similar could be done here as well?